pkg/controller/datavolume-controller_test.go

```diff
@@ -90,6 +90,90 @@ var _ = Describe("Datavolume controller reconcile loop", func() {
+		Expect(pvc.Spec.AccessModes[0]).To(Equal(corev1.ReadOnlyMany))
+	})
+
+	It("Should set params on a PVC from correct storageProfile when import DV has no accessMode", func() {
+		scName := "testStorageClass"
+
+		importDataVolume := newImportDataVolume("test-dv")
+		importDataVolume.Spec.PVC.StorageClassName = &scName
+		storageClass := createStorageClass(scName, nil) //, map[string]string{AnnDefaultStorageClass: "true"}
+		storageProfile := createStorageProfile(scName, []corev1.PersistentVolumeAccessMode{corev1.ReadOnlyMany}, corev1.PersistentVolumeBlock)
+		defaultStorageClass := createStorageClass("defaultSc", map[string]string{AnnDefaultStorageClass: "true"})
+		defaultStorageProfile := createStorageProfile("defaultSc", []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}, corev1.PersistentVolumeFilesystem)
+
+		reconciler = createDatavolumeReconciler(defaultStorageClass, storageClass, storageProfile, defaultStorageProfile, importDataVolume)
+
+		_, err := reconciler.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}})
+		Expect(err).ToNot(HaveOccurred())
+		pvc := &corev1.PersistentVolumeClaim{}
+		err = reconciler.client.Get(context.TODO(), types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}, pvc)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pvc.Name).To(Equal("test-dv"))
+
+		Expect(len(pvc.Spec.AccessModes)).To(BeNumerically("==", 1))
+		Expect(pvc.Spec.AccessModes[0]).To(Equal(corev1.ReadOnlyMany))
+		Expect(*pvc.Spec.VolumeMode).To(Equal(corev1.PersistentVolumeBlock))
+	})
+
+	It("Should set params on a PVC from default storageProfile when import DV has no storageClass and no accessMode", func() {
+		scName := "testStorageClass"
+		importDataVolume := newImportDataVolume("test-dv")
+		storageClass := createStorageClass(scName, map[string]string{AnnDefaultStorageClass: "true"})
+		storageProfile := createStorageProfile(scName, []corev1.PersistentVolumeAccessMode{corev1.ReadOnlyMany}, corev1.PersistentVolumeBlock)
+		anotherStorageProfile := createStorageProfile("anotherSp", []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}, corev1.PersistentVolumeFilesystem)
+
+		reconciler = createDatavolumeReconciler(storageClass, storageProfile, anotherStorageProfile, importDataVolume)
+
+		_, err := reconciler.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}})
+		Expect(err).ToNot(HaveOccurred())
+		pvc := &corev1.PersistentVolumeClaim{}
+		err = reconciler.client.Get(context.TODO(), types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}, pvc)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pvc.Name).To(Equal("test-dv"))
+
+		Expect(len(pvc.Spec.AccessModes)).To(BeNumerically("==", 1))
+		Expect(pvc.Spec.AccessModes[0]).To(Equal(corev1.ReadOnlyMany))
+		Expect(*pvc.Spec.VolumeMode).To(Equal(corev1.PersistentVolumeBlock))
```

Comment:
How do we handle DVs that do not mention a storage class?  The code now needs to resolve the default sc in order to look up the correct volume mode and access mode.  Since we are effectively deciding which storage class will be used, should we create the PVC with an explicit storage class?

Reply:
I need to check that and provide some tests.

The main question is:
- Do we support profiles when storage class is not present on DV? If yes, then we should add an explicit storageclass (default probably) to the PVC spec.

Reply:
The code actually just uses default storage class, and applies recommendations from StorageProfile for default SC.
Test for it is here:
https://github.com/kubevirt/containerized-data-importer/blob/70c58ecc45c5aa4951ea5d2bbaa127c29fa1f400/pkg/controller/datavolume-controller_test.go#L156

Reply:
DONE